Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -99,6 +119,63 @@ internal SFSession GetSession(string connStr)
             }
             return null;
         }
+
+        private SFSession NewSession(String connectionString, SecureString password)
+        {
+            s_logger.Debug("SessionPool::NewSession");
+            try
+            {
+                var session = new SFSession(connectionString, password);
+                session.Open();
+                return session;
+            }
+            catch (Exception e)
+            {
+                // Otherwise when Dispose() is called, the close request would timeout.
+                if (e is SnowflakeDbException)
+                    throw;
+                throw new SnowflakeDbException(
+                    e,
+                    SnowflakeDbException.CONNECTION_FAILURE_SSTATE,
+                    SFError.INTERNAL_ERROR,
+                    "Unable to connect. " + e.Message);
+            }
+        }
+
+        private Task<SFSession> NewSessionAsync(String connectionString, SecureString password, CancellationToken cancellationToken)
+        {
+            s_logger.Debug("SessionPool::NewSessionAsync");
+            try
+            {
+                var session = new SFSession(connectionString, password);
+                return session
+                    .OpenAsync(cancellationToken)
+                    .ContinueWith(previousTask =>
+                    {
+                        if (previousTask.IsFaulted && previousTask.Exception != null)
+                            throw previousTask.Exception;
+
+                        if (previousTask.IsFaulted)
+                            throw new SnowflakeDbException(
+                                SnowflakeDbException.CONNECTION_FAILURE_SSTATE,
+                                SFError.INTERNAL_ERROR,
+                                "Failure while opening session async");
+
+                        return session;
+                    }, TaskContinuationOptions.NotOnCanceled);
+            }
+            catch (Exception e)
```

Comment:
I think we don't need try .. catch here. The only exception catched here can be from "new SFSession", but why to cover it?

Reply:
It's was to ensure that anything from new will be thrown as SnowflakeDbException but you're right - new Session done in SnowflakeDbConnection was not covered with try catch so to keep same logic I'll remove it